### PR TITLE
57 bug multiple model ids with mixed tasks raise unexpected exception

### PR DESCRIPTION
**Pull request by AJDERS**
This PR fixes issue #57 .

In case of an invalid combination of `model_id` and `task` a warning is logged.

### CODE REVIEW

File: src/aiai_eval/evaluator.py [hunk 1]
```diff
@@ -149,10 +153,16 @@ def evaluate(
...
+                try:
+                    self._evaluate_single(
+                        task_config=task_config,
+                        model_id=m_id,
+                    )
+                except InvalidArchitectureForTask:
+                    logger.warning(
+                        f"Skipping evaluation of {m_id} on {task_config.pretty_name} "
+                        "as the architecture is not supported by the task."
+                    )
```
Comment (saattrupdan) [inline src/aiai_eval/evaluator.py:165]: There's already a try-except block within `_evaluate_single`, which currently captures `InvalidEvaluation` - this should probably be moved in there.

We could _either_ catch separate exceptions within that method, _or_ make a base Exception class from which all of our custom Exceptions inherit from, and catch that exception in `_evaluate_single`.

**Review by saattrupdan — CHANGES_REQUESTED** [review]

File: src/aiai_eval/evaluator.py (hunk 1 above)
Comment (AJDERS) [reply src/aiai_eval/evaluator.py]: I did the former.

File: src/aiai_eval/evaluator.py [hunk 2]
```diff
@@ -244,12 +238,18 @@ def _evaluate_single(
...
+                logger.info(
+                    f"{model_id} could not be evaluated on {task_config.pretty_name}. "
+                    "Skipping."
+                )
+                logger.debug(f'The error message was "{e}".')
+            elif type(e) == InvalidArchitectureForTask:
+                logger.warning(
+                    f"Skipping evaluation of {model_id} on {task_config.pretty_name} "
+                    "as the architecture is not supported by the task."
+                )
```
Comment (saattrupdan) [inline src/aiai_eval/evaluator.py:251]: Splitting these up in separate except blocks is a bit cleaner:

```suggestion
        except InvalidEvaluation:
            logger.info(
                f"{model_id} could not be evaluated on {task_config.pretty_name}. "
                "Skipping."
            )
            logger.debug(f'The error message was "{e}".')
        except InvalidArchitectureForTask:
            logger.warning(
                f"Skipping evaluation of {model_id} on {task_config.pretty_name} "
                "as the architecture is not supported by the task."
            )
```

Comment (saattrupdan) [reply src/aiai_eval/evaluator.py:251]: Also, why is the second exception handling a warning, with the first one being an info?

File: tests/test_evaluator.py [hunk 3]
```diff
@@ -96,13 +94,17 @@ def test_evaluate_single_raise_exception_model_not_found(
...
     ):
-        with pytest.raises(InvalidArchitectureForTask):
+        with caplog.at_level(logging.WARNING):
             evaluator._evaluate_single(
                 task_config=task_config, model_id=existing_model_id
             )
+        assert (
+            f"Skipping evaluation of {existing_model_id} on {task_config.pretty_name} "
+            "as the architecture is not supported by the task."
+        ) in caplog.text
```
Comment (saattrupdan) [inline tests/test_evaluator.py:107]: Maybe change "in" to "=="?

**Review by saattrupdan — CHANGES_REQUESTED** [review]

**Comment by AJDERS** [comment]
Closing this as solving conflicts is slower than doing it from the start once more.

**Pull request closed without merging**